index.ts

```diff
@@ -0,0 +1,476 @@
+// Exported native API for @smogon/data
+//
+// DISCLAIMER:
+//
+//   Not meant to replace Dex/BattleDex: consumers can delegate to this native API
+//   and put their own layer on top which can extend or alter the functionality.
+//
+// CONSIDERATIONS:
+//
+//	  - Approach: async APIs are moved to top level so that data load can be done
+//	    once and then the rest of the APIs aren't cluttered with `await` everywhere.
+//	    Furthermore, individual pieces can be loaded (just `Abilities` vs. all data
+//	    types) if desired.
+//	  - `getByID` APIs are important for removing the `toID` call in `get` methods
+//	    for when we already know through typechecking that we have an ID.
+//	  - Descriptions are UI only and can be large so aren't included in the Dex by
+//	    default, but they may be loaded in bulk asynchronously for the desired
+//	    languages.
+//   - The aliases.json data is used internally (to implement get/getByID etc) but
+//     not exported.
+//   - `forGenSync` methods can be added everywhere `forGen` exists as a non-Web
+//     convenience feature, though simply using `await` should be preferred.
+//
+//  *** TODO ***:
+//
+//   - 'Species' name is problematic as it is plural
+//   - How should 'letsgo'/'stadium' (mods) be handled? Option: use `forMod`
+//		 instead of `forGen`, and add a `genToMod(gen: Generation)` helper method
+//		 (plus overloads?)
+//	  - Instead of implemented `next()` directly, should `keys/values/entries` style
+//	    Iterator methods be written?
+//
+
+// data.ts
+
+type ID = '' | string & {__isID: true};
+toID(s: string): ID;
+
+type Generation = 1 | 2 | 3 | 4 | 5 | 6 | 7;
+
+type DataKind = 'Ability' | 'Move' | 'Item' | 'Species';
+type Nonstandard = 'Glitch' | 'Past' | 'Future' | 'CAP' | 'LGPE' | 'Pokestar' | 'Custom';
+interface Data {
+	id: ID;
+	kind: DataKind;
+	name: string;
+	num: number;
+	gen: Generation;
+	isUnreleased?: boolean;
+	isNonstandard: Nonstandard | null;
+}
+
+// natures.ts
+
+type NatureName = 'Adamant' | 'Bashful' | 'Bold' | 'Brave' | 'Calm' | 'Careful' | 'Docile' | 'Gentle' |
+	'Hardy' | 'Hasty' | 'Impish' | 'Jolly' | 'Lax' | 'Lonely' | 'Mild' | 'Modest' | 'Naive' | 'Naughty' |
+	'Quiet' | 'Quirky' | 'Rash' | 'Relaxed' | 'Sassy' | 'Serious' | 'Timid';
+
+class Natures {
+	get(name: string): Nature | undefined;
+	next(): [ID, Nature] | true;
```

Comment:
I don't see the point in returning `[ID, <T implements Data>]` when the `id` is available